main.go

```diff
@@ -17,33 +20,96 @@ import (
 	"github.com/gocardless/draupnir/store"
 	"github.com/gocardless/draupnir/version"
 	"github.com/gorilla/mux"
-	"github.com/kelseyhightower/envconfig"
 	"github.com/oklog/run"
+	"github.com/pkg/errors"
 )
 
 type Config struct {
-	Port                   int    `required:"true"`
-	InsecurePort           int    `required:"false" default:"8080"`
-	DatabaseUrl            string `required:"true" split_words:"true"`
-	DataPath               string `required:"true" split_words:"true"`
-	Environment            string `required:"false"`
-	SharedSecret           string `required:"true" split_words:"true"`
-	OauthRedirectUrl       string `required:"true" split_words:"true"`
-	OauthClientId          string `required:"true" split_words:"true"`
-	OauthClientSecret      string `required:"true" split_words:"true"`
-	TlsCertificatePath     string `required:"true" split_words:"true"`
-	TlsPrivateKeyPath      string `required:"true" split_words:"true"`
-	TrustedUserEmailDomain string `required:"true" split_words:"true"`
-	SentryDsn              string `required:"false" split_words:"true"`
+	Port                   int    `toml:"port"`
+	InsecurePort           int    `toml:"insecure_port"`
+	DatabaseUrl            string `toml:"database_url"`
+	DataPath               string `toml:"data_path"`
+	Environment            string `toml:"environment"`
+	SharedSecret           string `toml:"shared_secret"`
+	OauthRedirectUrl       string `toml:"oauth_redirect_url"`
+	OauthClientId          string `toml:"oauth_client_id"`
+	OauthClientSecret      string `toml:"oauth_client_secret"`
+	TlsCertificatePath     string `toml:"tls_certificate_path"`
+	TlsPrivateKeyPath      string `toml:"tls_private_key_path"`
+	TrustedUserEmailDomain string `toml:"trusted_user_email_domain"`
+	SentryDsn              string `toml:"sentry_dsn"`
+}
+
+const ConfigFilePath = "/etc/draupnir/config.toml"
+
+func loadConfig(path string) (Config, error) {
+	var config Config
+	file, err := os.Open(path)
+	if err != nil {
+		return config, errors.Wrap(err, fmt.Sprintf("No configuration file found at %s", ConfigFilePath))
+	}
+
+	_, err = toml.DecodeReader(file, &config)
+	if err != nil {
+		return config, errors.Wrap(err, "Could not parse configuration file")
+	}
+
+	return config, validateConfig(config)
+}
+
+func validateConfig(config Config) error {
```

Comment:
I thin this is horrible but I couldn't come up with a better way to validate that you've provided all configuration options.

Reply:
I think this is horrible too. Why are you like this.

```go
func (cfg Config) Validate() error {
	cfgValue := reflect.ValueOf(&cfg).Elem()
	emptyFields := []string{}

	for i := 0; i < cfgValue.NumField(); i++ {
		field := cfgValue.Field(i)
		empty := reflect.Zero(field.Type())
		if reflect.DeepEqual(field.Interface(), empty.Interface()) {
			emptyFields = append(emptyFields, cfgValue.Type().Field(i).Name)
		}
	}

	if len(emptyFields) == 0 {
		return nil
	}

	return fmt.Errorf("Missing required fields: %v", emptyFields)
}
```

Reply:
Ew that's also horrible. But at least shorter.

Reply:
Actually this approach doesn't even work, because we have some clearly-optional fields (sentry_dsn). We might have to add a `required` struct tag and use that. Ew ew ew.

Reply:
Yeah I was trolling a bit. It's also gross, but the best you get in Go.

We could add a required struct tag, or alternatively pass in fields that are optional.